include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_dual_splitting.cpp

```diff
@@ -696,37 +696,13 @@ TimeIntBDFDualSplitting<dim, Number>::viscous_step()
   dealii::Timer timer;
   timer.restart();
 
-  if(this->param.viscous_problem())
+  // in case we need to iteratively solve a linear or nonlinear system of equations
+  if(this->param.viscous_problem() or this->param.implicit_convective_problem())
   {
-    // if a variable viscosity is used: update
-    // viscosity model before calculating rhs_viscous
-    if(this->param.viscosity_is_variable())
-    {
-      dealii::Timer timer_viscosity_update;
-      timer_viscosity_update.restart();
-
-      // extrapolate velocity to time t_n+1 and use this velocity field to
-      // update the viscosity model (to recalculate the variable viscosity)
-      VectorType velocity_extrapolated(velocity[0]);
-      velocity_extrapolated = 0;
-      for(unsigned int i = 0; i < velocity.size(); ++i)
-        velocity_extrapolated.add(this->extra.get_beta(i), velocity[i]);
```

Comment:
@richardschu This code was somewhat redundant to the extrapolation of the velocity used as initial guess in the viscous step. I refactored the code (by changing the interface of the function `rhs_viscous` and storing the respective velocity vector) so that we can avoid this redundancy. The code in the function `viscous_step()` is now structured very similarly to the pressure-correction scheme.